website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -65,6 +65,43 @@ The most common types are:
  - `Gauge` Value can be increased and decreased as needed.
 5. `_total` suffix should be used for accumulating count. If your metrics has labels with high cardinality, like `pod`/`container` it usually means that you can aggregate it more, thus it will not require `_total` suffix.
 
+#### Recommended Metrics
+
+{{% alert title="Warning" color="warning" %}}
+The following operator health metric is new and may be subjected to change 
```

Comment:
Could we make this a copy of the note which we have below.

"The metrics below are are the suggested values. If you would prefer to follow any other level of granularity to indicate health statuses, a different metric name MUST be used, and it should be clearly documented such that an external observer can infer the indicator values accordingly."

I do not see a value in just adding that these metrics are subject to change. Imo, it is practically impossible for us to cover all use cases and define a standardized set of metrics that fits all. As mentioned before, as an operator author, I could have "success/failure/critical" as three sets based on my operator's lifecycle, where as someone else could have introduced a new terminology with "success/warning/degraded/critical" etc based on their operator's state, which IMO is not wrong. Though the idea here is to suggest best practices, we can't practically create a standard that serves all the purposes. We could have operator authors suggesting to add "n" number of levels which suite their use case, but it is not practical for us to modify the set based on that. Even if we decide to do so, what is the basis with which we will consider that a suggested health status is useful to be added to this set?

Imo, as experts our task should be to provide necessary docs/helpers to encourage operator authors to set up alerts with an example that defines a generic naming convention and status levels (that is reasonable and less confusing) than us working on defining a specific set with its respective definitions.

> I think a really good way to do that would be to implement something where the author just has to provide a function or hook and then the library does all the heavy lifting to register the metric, and set it to the right 0/1/2 value.

This seems to be generic enough to not only be used in operators but anywhere else. Wrappers like these could be a great addition to prometheus itself than to KB or SDK.

Reply:
@varshaprasad96 I suggested the "subject to change" wording because it seems like it would be good to get some feedback about this from a wider audience before we declare it a best practice. What it seems like we're saying is:
1. All operators should _at least_ specify _this_ metric.
2. We don't have enough feedback to say for sure this specific implementation is a best practice, so we _may_ change some of the specifics (e.g. how to choose the name label, what the various states are, how to convey those states, etc). What we probably will _not_ change is the overall concept of this metric.
3. If you think this metric is not granular enough, there's nothing stopping you from populating a separate metric. Just don't hijack this one.

Maybe we could add a little more to this warning message along the lines of item (2), and maybe add a call to action about providing feedback?

Reply:
Updated